File: DemoServer/AdditionalLanguages/python/queries/projectingIndividualFields/projectingIndividualFields.py
from typing import List
from models import Company

#region Usings
from ravendb import QueryData
#endregion

#region Demo
class CompanyDetails:
    def __init__(self, company_name: str = None, city: str = None, country: str = None):
        self.company_name = company_name
        self.city = city
        self.country = country

#endregion

def run(self, run_params=None) -> List[CompanyDetails]:
    #region Demo
    with self.document_store_holder.store().open_session() as session:
        #region Step_1
        projected_query = session.query(object_type=Company)
        #endregion
        
        #region Step_2
        projected_query = projected_query.select_fields_query_data(
            CompanyDetails,
            QueryData(
                ["Name", "Address.City", "Address.Country"],
                ["company_name", "city", "country"],
            ),
        )
        #endregion

        #region Step_3
        projected_results = list(projected_query)
        #endregion
    #endregion
    return projected_results

File: DemoServer/AdditionalLanguages/php/queries/projectingIndividualFields/ProjectingIndividualFields.php
<?php

namespace RavenDB\Demo\queries\projectingIndividualFields;

//region Usings
use RavenDB\Documents\Queries\QueryData;
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;
use RavenDB\Demo\common\models\Company;

class ProjectingIndividualFields
{
    public function __invoke(): array
    {
        $projectedResults = [];

        //region Demo
        $session = DocumentStoreHolder::getStore()->openSession();
        
        try {
            //region Step_1
            $projectedQuery = $session->query(Company::class)
            //endregion
                //region Step_2
                    ->selectFields(CompanyDetails::class,
                        new QueryData(
                            [ "Name", "Address.City", "Address.Country" ],
                            [ "CompanyName", "City", "Country" ]));
                //endregion

            //region Step_3
            $projectedResults = $projectedQuery->toList();
            //endregion
        } finally {
            $session->close();
        }
        //endregion
        return $projectedResults;
    }
}
